src/Language/PureScript/Make.hs

```diff
@@ -160,7 +178,7 @@ make MakeActions{..} ms = do
     (_, env') <- lift . runCheck' env $ typeCheckModule Nothing m
     go env' ms'
   go env ((True, m@(Module coms moduleName' _ exps)) : ms') = do
-    lift $ progress $ "Compiling " ++ runModuleName moduleName'
+    when (isJust formatter) $ lift $ progress $ fromJust formatter $ runModuleName moduleName'
```

Comment:
You can avoid `when (isJust ...) $ fromJust ...` by using `Data.Foldable.for_` instead, with the `Foldable Maybe` instance:

`when (isJust x) $ f $ fromJust x == for_ x f`

This might be easier to read if you define `whenJust` as a specialization of  `for_`, so `whenJust :: (Applicative f) => Maybe a -> (a -> f b) -> f ()`